src/io/flutter/view/InspectorTreeUI.java

```diff
@@ -0,0 +1,647 @@
+            if (parentPath != null) {
+              if (parentPath.getParentPath() != null || tree.isRootVisible()) {
+                final int parentRow = tree.getRowForPath(parentPath);
+                tree.scrollRowToVisible(parentRow);
+                tree.setSelectionInterval(parentRow, parentRow);
+              }
+            }
+          }
+          else {
+            tree.collapseRow(selectionRow);
+          }
+        }
+      }
+    });
+  }
+
+  private abstract static class TreeUIAction extends AbstractAction implements UIResource {
+  }
+
+  @Override
+  protected int getRowX(int row, int depth) {
+    if (isCustomIndent()) {
+      final int off = tree.isRootVisible() ? 8 : 0;
+      return 8 * depth + 8 + off;
+    } else {
+      return super.getRowX(row, depth);
+    }
+  }
+
+  @Override
+  protected void paintHorizontalPartOfLeg(final Graphics g,
+                                          final Rectangle clipBounds,
+                                          final Insets insets,
+                                          final Rectangle bounds,
+                                          final TreePath path,
+                                          final int row,
+                                          final boolean isExpanded,
+                                          final boolean hasBeenExpanded,
+                                          final boolean isLeaf) {
+    if (path.getPathCount() < 2) {
+      // We could draw lines for nodes with a single child but we omit them
+      // to more of an emphasis of lines for nodes with multiple children.
+      return;
+    }
+    if (path.getPathCount() >= 2) {
+      final Object treeNode = path.getPathComponent(path.getPathCount() - 2);
+      if (treeNode instanceof DefaultMutableTreeNode) {
+        DefaultMutableTreeNode node = (DefaultMutableTreeNode) treeNode;
+        if (node.getChildCount() < 2) {
+          return;
+        }
+      }
+    }
+    boolean dashed = false;
+    final DiagnosticsNode diagnosticsNode = maybeGetDiagnostic((DefaultMutableTreeNode)path.getLastPathComponent());
+    if (diagnosticsNode != null) {
+      if (diagnosticsNode.isProperty()) {
+        // Intentionally avoid ever drawing lines for properties as we need to
+        // distinguish them from other nodes. See the DiagnosticsNode class in
+        // Flutter which applies the same concept rendering properties inline
+        // as part of ascii art tree display.
+        return;
+      }
+      // This also consistent with the ascii art tree display where offstage
+      // nodes are rendered using dashed lines.
+      if (diagnosticsNode.getStyle() == DiagnosticsTreeStyle.offstage) {
+        dashed = true;
+      }
+    }
+
+    final int depth = path.getPathCount() - 1;
+    if((depth == 0 || (depth == 1 && !isRootVisible())) &&
+       !getShowsRootHandles()) {
+      return;
+    }
+
+    final int clipLeft = clipBounds.x;
```

Comment:
In terms of simplifying this, would any of the `Rectangle` / `Shape` classes help?

Reply:
Done. Switched to using the Rectangle class's intersects method.